### ISSUE TITLE
[Feature Request] - best signal strenght with highest uplink speeds combined

### ISSUE THREAD

**Issue by Abysium** [TrilbyWhite/swifer#2]
Hey

When swifer connects to a network it only measures the signal strenght, but that is in no way a valid marker of uplink strenght behind the AP.

Would it be possible to implement a probe mode that would make swifer compile a table of all open networks it sees, then connect to each one, run a quick speedtest, note results and deduce which one is optimal?

---

I have some more ideas and questions that i would like to go over with you if you are interested, I would have mailed them to you by now, but I cant seem to be able to find your email address anywhere. Please send a mail to 1337joze@gmail.com if interested.

Goodday

**Comment by TrilbyWhite** [comment]
Thanks for the request.  Unfortunately this first one I don't see as practical for two reasons.

First, just getting the network speed itself isn't trivial - I've considered doing this with the connected network to trigger a "reconnect" to another known network.

Second, connecting, probing, then disconnecting to many networks could take a _long_ time as a new dhcp would have to be run on each one.

I would support any effort on an external tool to acheive this, and I'd happily tailor swifer to interact well with such tools.  One way I could see this happening is swifer modes to dump a list of available networks, and command line parameters to connect to a specified network.  Then the external tool could even be a bash script something like the following hypothetical:[code]for network in $(swifer get-open); do
  swifer connect $network
  #check speed where
  [[ $speed -gt $threshold ]] && break;
done[/code]

This check, however, will not make it into swifer itself.

The other ideas can certainly be shared here.

**Comment by Abysium** [comment]
Yeah, I understand that probing speeds for each network could be relatively
slow, thats why i thought it could be a special option, not set by default.

Now for the next bit, could swifer be configured to provide uplink to a
second network adapter that would be configured as an AP?

I am asking this, becouse I am trying figure out a way to provide wireless
access on a mountain cabin, the way im currently invisioning it, I would
have a dedicated laptop with 2 wifi adapters, one using a yagi to get
signal from the valley bellow, and the other bridged to it to provide an AP
on the spot.

Since this would be a long term setup (4 months), it would be ideal if it
were as automatic as possible.

2013/4/19 Jesse McClure notifications@github.com

**Comment by TrilbyWhite** [comment]
That's an interesting idea, but it seems other tools already exist that are tailored more towards those purposes.

Swifer has always been a quick way to mostly just automate manual connection steps that would have to be often repeated when moving between access points.  In fact I wrote it first to just use on my own netbook as I was out and about.

There are other highly capable tools like Network Manager or Netcfg that will do exactly what you want.  Or even just using iw/iwconfig to do it manually as this would be set up once and run for several months.

I'm happy to develop swifer in new directions, but given your end goals, it may not be the right tool for the job.  NetworkManager or Netcfg would require less (if any) adjustment to get what you want.
